AGENTS.md

```
@@ -3,65 +3,338 @@
-**CLI**: `src/cli/`
+**Retrieval**: `src/retrieval/bm25.rs`, `src/retrieval/hybrid.rs`, `src/singularity_retrieval.rs`
+**CLI**: `src/cli/commands/query.rs`, `src/cli/commands/index_dir.rs`
 **State**: `plans/GOAP_STATE.md`, `plans/ACTIONS.md`
 
-## Skills (27 Total)
-**Core**: adr-creation, benchmarking-perf, debugging-reservoir, dist-channel-selection, drawio, git-workflow, github-ci-guardrails, goap-planning, memory-lifecycle-verification, npm-trusted-publishers, release-management, rust-development, testing-validation, turso-memory-verification
-**Swarm**: analysis-swarm, swarm-advanced-features, swarm-observability
-**Workflow**: learn, shell-script-quality, task-decomposition
-**Automation**: jules-orchestration, self-fix-loop, skill-creator, skill-evaluator, skill-memory-internal
-**TRIZ**: triz-analysis, triz-solver
+## Skills (30 Total)
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)